### PR TITLE
Report error when instantiating CanvasKit network image

### PR DESCRIPTION
**Pull request by hterkelsen**
## Description

If there is an error when performing the HttpRequest for a CanvasKit network image, report it to the
framework for proper handling.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/65537

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: lib/web_ui/lib/src/engine/canvaskit/image.dart [hunk 1]
```diff
@@ -115,7 +127,8 @@ class CkAnimatedImage implements ui.Image {
       );
       bytes = _skAnimatedImage.readPixels(imageInfo, 0, 0);
     } else {
-      final SkData skData = _skAnimatedImage.encodeToData(); //defaults to PNG 100%
+      final SkData skData =
+          _skAnimatedImage.encodeToData(); //defaults to PNG 100%
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/image.dart:131]: nit: add space after //

File: lib/web_ui/lib/src/engine/canvaskit/image.dart [hunk 2]
```diff
@@ -15,24 +15,33 @@ void skiaInstantiateImageCodec(Uint8List list, Callback<ui.Codec> callback,
...
+  Completer<ui.Codec?> completer = Completer<ui.Codec?>();
   chunkCallback?.call(0, 100);
   //TODO: Switch to using MakeImageFromCanvasImageSource when animated images are supported.
   html.HttpRequest.request(
     src,
     responseType: "arraybuffer",
   ).then((html.HttpRequest response) {
+    if (response.status != 200) {
+      completer.completeError(
+          'Network image request failed with status: ${response.status}');
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/image.dart:31]: Given the amount of confusion this is causing, I think it's worth elaborating on what might be going on here. Alternatively, we could have a wiki page that explains CORS and link to it from here.

**Review by yjbanov — COMMENTED** [review]
Is this testable?

File: lib/web_ui/lib/src/engine/canvaskit/image.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/canvaskit/image.dart]: Done.

File: lib/web_ui/lib/src/engine/canvaskit/image.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/canvaskit/image.dart]: CORS exceptions are caught in the onError handler below. This catches "successful" requests which aren't status 200. For CORS exceptions an error is dumped to the console explaining that the image to be downloaded didn't have CORS headers set.

**Review by yjbanov — APPROVED** [review]
LGTM is LGT CI

lgtm

**Review by yjbanov — APPROVED** [review]

**Pull request merged**